src/Sylius/Bundle/ResourceBundle/Event/ResourceControllerEvent.php

```diff
@@ -157,4 +173,14 @@ public function setErrorCode($errorCode)
     {
         $this->errorCode = $errorCode;
     }
+
+    /**
+     * Return the request configuration.
+     *
+     * @return RequestConfiguration
+     */
+    public function getRequestConfiguration()
+    {
+        return $this->requestConfiguration;
```

Comment:
Event has access to the request configuration, and through it the Request. For my use case this will allow access to a `parent_path` query parameter, allowing documents to be added at explicit points in a tree - this is required when creating a document from a tree browser
![treebrowse](https://cloud.githubusercontent.com/assets/530801/16712637/75f16b06-4685-11e6-8293-5ab881570705.png)